service/src/test/java/org/cbioportal/genome_nexus/service/enricher/IsoformAnnotationEnricherTest.java

```diff
@@ -0,0 +1,116 @@
+    public void enrichAnnotationWithSingleOverride() throws IOException
+    {
+        Map<String, VariantAnnotation> variantMockData = this.variantAnnotationMockData.generateData();
+
+        IsoformAnnotationEnricher enricher = new IsoformAnnotationEnricher(
+            "uniprot", "uniprot", this.ensemblService
+        );
+
+        // override canonical transcripts with just one matching transcript
+        Set<String> isoformOverrides = new HashSet<>();
+        isoformOverrides.add("ENST00000429201");
+        Mockito
+            .when(this.ensemblService.getCanonicalTranscriptIdsBySource("uniprot"))
+            .thenReturn(isoformOverrides);
+
+
+        VariantAnnotation annotation = variantMockData.get("3:g.14106026_14106037del");
+        enricher.enrich(annotation);
+        List<TranscriptConsequence> canonicalTranscripts =
+            annotation
+                .getTranscriptConsequences()
+                .stream()
+                .filter(t -> t.getCanonical() != null && t.getCanonical().equals("1"))
+                .collect(Collectors.toList());
+
+        assertEquals(
+            1,
+            canonicalTranscripts.size()
+        );
+
+        assertEquals(
+            "ENST00000429201",
+            canonicalTranscripts.get(0).getTranscriptId()
+        );
+    }
+
+    @Test
+    public void enrichAnnotationWithMultipleOverride() throws IOException
+    {
+        Map<String, VariantAnnotation> variantMockData = this.variantAnnotationMockData.generateData();
+
+        IsoformAnnotationEnricher enricher = new IsoformAnnotationEnricher(
+            "mskcc", "mskcc", this.ensemblService
+        );
+
+        // override canonical transcripts with just one matching transcript
+        Set<String> isoformOverrides = new HashSet<>();
+        isoformOverrides.add("ENST00000532753");
+        isoformOverrides.add("ENST00000532924");
+        isoformOverrides.add("ENST00000532880");
+        Mockito
+            .when(this.ensemblService.getCanonicalTranscriptIdsBySource("mskcc"))
+            .thenReturn(isoformOverrides);
+
+
+        VariantAnnotation annotation = variantMockData.get("3:g.14106026_14106037del");
+        enricher.enrich(annotation);
+        List<TranscriptConsequence> canonicalTranscripts =
+            annotation
+                .getTranscriptConsequences()
+                .stream()
+                .filter(t -> t.getCanonical() != null && t.getCanonical().equals("1"))
+                .collect(Collectors.toList());
+
+        assertEquals(
+            3,
+            canonicalTranscripts.size()
+        );
+
+        assertEquals(
+            "ENST00000532753",
+            canonicalTranscripts.get(0).getTranscriptId()
+        );
+
+        assertEquals(
+            "ENST00000532880",
+            canonicalTranscripts.get(1).getTranscriptId()
+        );
+
+        assertEquals(
+            "ENST00000532924",
+            canonicalTranscripts.get(2).getTranscriptId()
+        );
+    }
+}
```

Comment:
maybe a missing test : enrichAnnotationWithZeroOverrides ?

Reply:
sounds good, I can add that one too.